nodebuilder/core/config.go

```diff
@@ -0,0 +1,57 @@
+package core
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Config combines all configuration fields for managing the relationship with a Core node.
+type Config struct {
+	IP       string
+	RPCPort  string
+	GRPCPort string
+}
+
+// DefaultConfig returns default configuration for managing the
+// node's connection to a Celestia-Core endpoint.
+func DefaultConfig() Config {
+	return Config{
+		IP:       "0.0.0.0",
+		RPCPort:  "0",
+		GRPCPort: "0",
+	}
+}
+
+// ValidateBasic performs basic validation of the config.
+func (cfg *Config) ValidateBasic() error {
+	_, err := sanityCheckIP(cfg.IP)
+	if err != nil {
+		return err
+	}
```

Comment:
we should set cfg.IP to return value here because `sanityCheckIP` doesn't just sanity check, it also trims leading/trailing characters. We should probs rename to `sanitizeIP` or something.